crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,18 +83,30 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    let line_length: u16 = checker.settings.line_length.into();
+                    if !locator.contains_line_break(source_range)
+                        && locator.full_line(source_range.start()).trim_end().len()
+                            <= ((line_length as usize) - 2)
```

Comment:
This operation compares the byte-length instead of the text width. The byte and text width are identical for ASCII characters but non-ASCII characters either require multiple-bytes to represent them. Then there are Emojis and east-asian characters that take multiple columns when rendered on screen. Last, there's also the tab character which uses up to `tab-width` columns when shown in the editor. 

 Luckily, Ruff has the infrastructure to compute the visual width of a line. Take a look at the `LineWidthBuilder` and its usages to compute the width of a line.

https://github.com/astral-sh/ruff/blob/84979f96736aefca6d19f656e68600a92690c737/crates/ruff_linter/src/line_width.rs#L125-L133

Reply:
Ah, I _did_ see `LineWidthBuilder`, but decided I probably didn't need something quite so fancy for my purposes here. My mistake! :)